operators/src/processing/raster_kernel/mod.rs

```diff
@@ -0,0 +1,737 @@
+
+        // let result_stream = processor.query(query_rect, &query_ctx).await.unwrap();
+
+        let colorizer = Colorizer::linear_gradient(
+            vec![
+                (0.0, RgbaColor::white()).try_into().unwrap(),
+                (255.0, RgbaColor::black()).try_into().unwrap(),
+            ],
+            RgbaColor::transparent(),
+            RgbaColor::pink(),
+        )
+        .unwrap();
+
+        let bytes = raster_stream_to_png_bytes(
+            processor,
+            query_rect,
+            query_ctx,
+            360,
+            180,
+            None,
+            Some(colorizer),
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(
+            bytes,
+            include_bytes!("../../../../test_data/wms/gaussian_blur.png")
+        );
+
+        // Use for getting the image to compare against
+        // save_test_bytes(&bytes, "gaussian_blur.png");
+    }
+
+    #[tokio::test]
+    async fn test_ndvi_sobel_filter() {
+        let mut exe_ctx = MockExecutionContext::test_default();
+        let ndvi_id = add_ndvi_dataset(&mut exe_ctx);
+
+        let operator = RasterKernel {
+            params: RasterKernelParams {
+                kernel: RasterKernelMethod::Convolution {
+                    matrix: vec![vec![1., 0., -1.], vec![2., 0., -2.], vec![1., 0., -1.]],
+                },
+            },
+            sources: SingleRasterSource {
+                raster: GdalSource {
+                    params: GdalSourceParameters { data: ndvi_id },
+                }
+                .boxed(),
+            },
+        }
+        .boxed()
+        .initialize(&exe_ctx)
+        .await
+        .unwrap();
+
+        let processor = operator.query_processor().unwrap().get_u8().unwrap();
+
+        let query_rect = RasterQueryRectangle {
+            spatial_bounds: SpatialPartition2D::new((-180., 90.).into(), (180., -90.).into())
+                .unwrap(),
+            time_interval: TimeInstance::from(DateTime::new_utc(2014, 1, 1, 0, 0, 0)).into(),
+            spatial_resolution: SpatialResolution::one(),
+        };
+        let query_ctx = MockQueryContext::test_default();
+
+        // let result_stream = processor.query(query_rect, &query_ctx).await.unwrap();
+
+        let colorizer = Colorizer::linear_gradient(
+            vec![
+                (0.0, RgbaColor::white()).try_into().unwrap(),
+                (255.0, RgbaColor::black()).try_into().unwrap(),
+            ],
+            RgbaColor::transparent(),
+            RgbaColor::pink(),
+        )
+        .unwrap();
+
+        let bytes = raster_stream_to_png_bytes(
+            processor,
+            query_rect,
+            query_ctx,
+            360,
+            180,
+            None,
+            Some(colorizer),
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(
+            bytes,
+            include_bytes!("../../../../test_data/wms/sobel_filter.png")
```

Comment:
didn't you say that this wasn't actually a sobel_filter?

Reply:
[b6b3237](https://github.com/geo-engine/geoengine/pull/644/commits/b6b323705d0d990d2ce33577c0f17192939cefcb)